### PR TITLE
Phase2-hgx344 Resolve the issue of non-matching DetIDs from DDD and DD4hep for HGCal versions in D98 and D99

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Resolve the issue of non-matching DetIDs from DDD and DD4hep for HGCal versions in D98 and D99

#### PR validation:

Tested using the scripts in Geometry/HGCalCommonData/testpython

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include <functional>
 #include <numeric>
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:22]: @bsunanda , likely DEBUG should be disabled.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
type bug-fix

**Comment by civanch** [comment]
This PR should resolve #42224

**Comment by srimanob** [comment]
+Upgrade

Change is on Phase-2 DD4hep workflow only. Validation of Phase-2 DD4hep should be organized in the next pre-release, @cms-sw/pdmv-l2

**Comment by bsunanda** [comment]
+geometry

**Comment by perrotta** [comment]
+1
- Validation with CMSSW_13_3_0_pre2 is expected

**Pull request merged**